roles/openshift_storage_glusterfs/tasks/heketi_deploy_part2.yml

```diff
@@ -0,0 +1,109 @@
+---
+- name: Create heketi DB volume
+  command: "heketi-cli -s http://{{ openshift_storage_glusterfs_heketi_url }} --user admin --secret '{{ openshift_storage_glusterfs_heketi_admin_key }}' setup-openshift-heketi-storage --listfile {{ mktemp.stdout }}/heketi-storage.json"
+  register: setup_storage
+  failed_when: False
+
+# This is used in the subsequent task
+- name: Copy the admin client config
+  command: >
+    cp {{ openshift.common.config_base }}/master/admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
+  changed_when: False
+  check_mode: no
+
+# Need `command` here because heketi-storage.json contains multiple objects.
+- name: Copy heketi DB to GlusterFS volume
+  command: "{{ openshift.common.client_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig create -f {{ mktemp.stdout }}/heketi-storage.json -n {{ openshift_storage_glusterfs_namespace }}"
```

Comment:
@jarrpa, 

I believe oc_process could do the same thing here.  That way we move away from copying the kubeconfig step and running command module.  The modules attempt to idempotently handle the operations.

Something like this:
```yaml
- name: Create the gluster template 
  oc_obj:
     name: glustertemplate
     namespace: "{{ openshift_storage_glusterfs_namespace }}"
     files:
     - "{{ mktemp.stdout }}/heketi-storage.json"
     kind: template

-  name: Create all objects from the glustertemplate
   oc_process:
     namespace: "{{ openshift_storage_glusterfs_namespace }}"
     template_name: glustertemplate
     state: present
     create: True
     reconcile: True
```
I'd prefer the oc_obj but if that doesn't work you could try without the oc_obj:  
```yaml
- slurp:
    src: "{{ mktemp.stdout }}/heketi-storage.json"
  register: glusterobjects
-  name: Create all objects from the glustertemplate
   oc_process:
     namespace: "{{ openshift_storage_glusterfs_namespace }}"
     template_name: glustertemplate
     state: present
     create: True
     reconcile: True
     content: "{{ glusterobjects | b64decode }}"
```
Of course, I haven't tested this but we do similar operations in other places.